File: memory_bit(2).c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>
#include "gates.h"

int main()
{
    int i = 0, s = 0, a = 0, b = 0, c = 0, o = 0;

    while(i != 999)
    {
        printf("To remember: \n");
        fflush;
        scanf("%d", &i);

        printf("Set value: \n");
        fflush;
        scanf("%d", &s);

        a = NAND(i, s);
        printf("a: %d\n", a);
        b = NAND(a, s);
        printf("b: %d\n", b);
        c = NAND(o, b);
        printf("c: %d\n", c);
        o = NAND(a, c);
        printf("o: %d\n", o);

        printf("The output is: %d\n", o);
    }
}

File: py/memory_bit(2).py
def NAND(a, b):
    if a == 1 and b == 1:
        return 0
    else:
        return 1

def memory(i, s):
    a = 0
    b = 0
    c = 0
    o = 0
    while(1):
        a = NAND(i, s)
        b = NAND(a, s)
        c = NAND(o, b)
        o = NAND(a, c)

        print("Output bit: %d\n", o)
            

i = input("Enter bit to remember: ")
s = input("Set bit: ")

memory(i, s)
